### PR TITLE
Make block device allocation settings configurable

### PR DESCRIPTION
**Pull request by nkrinner**
Make block_device_allocate_retries and block_device_allocate_retries_interval
from the crowbar web ui.
See also https://bugzilla.suse.com/show_bug.cgi?id=937800

### CODE REVIEW

**Comment by jsuchome** [comment]
Looks good so far. I think if it works it's a way to go.

File: chef/data_bags/crowbar/migrate/nova/011_configure_block_device_allocation.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+def upgrade ta, td, a, d
+  a['block_device_allocation_retries'] = 60
```
Comment (tboerger) [inline chef/data_bags/crowbar/migrate/nova/011_configure_block_device_allocation.rb]: Normally we use `ta["block_device_allocation_retries"]` for assignment.

File: chef/data_bags/crowbar/migrate/nova/011_configure_block_device_allocation.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  a['block_device_allocation_retries_interval'] = 3
```
Comment (tboerger) [inline chef/data_bags/crowbar/migrate/nova/011_configure_block_device_allocation.rb]: Normally we use `ta["block_device_allocation_retries_interval"]` for assignment.

**Comment by nkrinner** [comment]
Removed the [WIP] tag from the pull request headline. Unrelated if this fixes the bug, this additional configuration option can get merged to the code. Will also submit against crowbar-openstack when this gets merged.

**Comment by rsalevsky** [comment]
+1

**Comment by MaximilianMeister** [comment]
+1 codewise

File: chef/data_bags/crowbar/bc-template-nova.json [hunk 3]
```diff
@@ -98,14 +98,16 @@
           "certfile": "",
           "keyfile": ""
         }
-      }
+      },
+      "block_device_allocate_retries": 60,
+      "block_device_allocate_retries_interval": 3
```
Comment (vuntz) [inline chef/data_bags/crowbar/bc-template-nova.json:104]: Would it make sense to create a new group for the options?

**Comment by vuntz** [comment]
@nkrinner in general, this must go in crowbar-openstack first, then we can merge the backport.

File: chef/data_bags/crowbar/bc-template-nova.json (hunk 3 above)
Comment (tboerger) [inline chef/data_bags/crowbar/bc-template-nova.json:104]: Sounds good to me

**Comment by nkrinner** [comment]
Closing this pr for now, as the discussion is happening in https://github.com/crowbar/crowbar-openstack/pull/39 and this one got messed up during rebasing.

**Pull request closed without merging**